### PR TITLE
Flag for enabling no-wait excludes

### PR DESCRIPTION
**Pull request by 09harsh**
# Description

*Please include a summary of the change and which issue is addressed. If this change resolves an issue, please include the issue number in the description.*
Added a new flag in the `AutomaticReplacementOptions` for using non blocking excludes. This solves issue #349 

## Type of change
* New feature (non-breaking change which adds functionality)

# Discussion

*Are there any design details that you would like to discuss further?*
No

# Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*
Added Unit test 

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

# Documentation

*Did you update relevant documentation within this repository?*
Yes

*If this change is adding new functionality, do we need to describe it in our user manual?*

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

# Follow-up

*Are there any follow-up issues that we should pursue in the future?*

*Does this introduce new defaults that we should re-evaluate in the future?*

### CODE REVIEW

File: fdbclient/admin_client.go [hunk 1]
```diff
@@ -237,7 +237,8 @@ func (client *cliAdminClient) ExcludeInstances(addresses []fdbtypes.ProcessAddre
 		return err
 	}
 
-	if version.HasNonBlockingExcludes() {
+	useNonBlockingExcludes := client.Cluster.GetUseNonBlockingExcludes()
+	if version.HasNonBlockingExcludes(useNonBlockingExcludes) {
```
Comment (johscheuer) [inline fdbclient/admin_client.go:241]:
```suggestion
	if version.HasNonBlockingExcludes( client.Cluster.GetUseNonBlockingExcludes()) {
```

File: api/v1beta1/foundationdb_version_test.go [hunk 2]
```diff
@@ -91,4 +92,43 @@ var _ = Describe("[api] FDBVersion", func() {
...
+	FWhen("checking if the version has support for non-blocking exclude commands", func() {
+		type testCase struct {
+			version                FdbVersion
+			useNonBlockingExcludes bool
+			expectedResult         bool
+		}
+
+		DescribeTable("should check the version supports non-blocking exclude commands",
+			func(tc testCase) {
+				Expect(tc.version.HasNonBlockingExcludes(tc.useNonBlockingExcludes)).To(BeEquivalentTo(tc.expectedResult))
```
[the remark this answers is not in the document]
Comment (09harsh) [reply api/v1beta1/foundationdb_version_test.go]: Yeah right, `Equal` would be a better option. It didn't clicked me while coding.

File: api/v1beta1/foundationdb_version.go [hunk 3]
```diff
@@ -155,11 +155,9 @@ func (version FdbVersion) HasZoneFaultToleranceInStatus() bool {
...
 // HasNonBlockingExcludes determines if a version has support for non-blocking
 // exclude commands.
-//
-// This is currently set to false across the board, pending investigation into
-// potential bugs with non-blocking excludes.
-func (version FdbVersion) HasNonBlockingExcludes() bool {
-	return false
+func (version FdbVersion) HasNonBlockingExcludes(useNonBlockingExcludes bool) bool {
+	return version.IsAtLeast(FdbVersion{Major: 6, Minor: 3, Patch: 5}) && useNonBlockingExcludes
+
```
Comment (johscheuer) [inline api/v1beta1/foundationdb_version.go:160]: Remove this newline.

**Review by johscheuer — APPROVED** [review]
LGTM 👍 I leave the final approval to @ltsampros

**Review by ltsampros — CHANGES_REQUESTED** [review]
LGTM but could you please ditch all the changes to the `go.sum` file? We can run `go mod tidy` separately.

**Pull request merged**